pykafka/producer.py

```diff
@@ -102,7 +124,32 @@ def __init__(self,
         self._retry_backoff_ms = retry_backoff_ms
         self._required_acks = required_acks
         self._ack_timeout_ms = ack_timeout_ms
-        self._batch_size = batch_size
+        self._max_queued_messages = max_queued_messages
+        self._min_queued_messages = min_queued_messages
+        self._linger_ms = linger_ms
+        self._block_on_queue_full = block_on_queue_full
+        self._worker_exception = None
+        self._worker_trace_logged = False
+        self._running = False
+        self.start()
+
+    def raise_worker_exceptions(fn):
+        """Decorator that raises exceptions encountered on worker threads
+        """
```

Comment:
[nitpick] If the docstring fits on one line, the closing triple-quote should be on that line too.